From: Lihua Yao <ylhuajnu@163.com>
Date: Wed, 15 Aug 2018 23:20:35 +0800
Subject: [PATCH] ALSA: ac97: fix unbalanced pm_runtime_enable
References: bnc#1012628
Patch-mainline: 4.18.7
Git-commit: 250ea7c5f56e350cdafebe6b87478b00db4f7af8

commit 250ea7c5f56e350cdafebe6b87478b00db4f7af8 upstream.

Runtime PM is enabled at ac97_bus_probe() and should be disabled
at ac97_bus_remove().

Fixes: 74426fbff66e ("ALSA: ac97: add an ac97 bus")
Signed-off-by: Lihua Yao <ylhuajnu@163.com>
Acked-by: Robert Jarzmik <robert.jarzmik@free.fr>
Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 sound/ac97/bus.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/ac97/bus.c b/sound/ac97/bus.c
index c267e44fefa0..83eed9d7f679 100644
--- a/sound/ac97/bus.c
+++ b/sound/ac97/bus.c
@@ -511,6 +511,8 @@ static int ac97_bus_remove(struct device *dev)
 	if (ret == 0)
 		ac97_put_disable_clk(adev);
 
+	pm_runtime_disable(dev);
+
 	return ret;
 }
 
-- 
2.18.0

